website/blog/2018-09-13-decorators.md

```diff
@@ -0,0 +1,133 @@
+
+The new decorators proposal is much more powerful: element decorators take an object which, other than changing the property descriptor, allows to change the key, the placement (`static`, `prototype` or `own`) and the kind (`field` or `method`) of the element. They can also create additional properties and define a function (a *finisher*) which is run on the decorated class.
+Class decorators take an object which contains the descriptors of every single class element, making it possible to modify them before creating the class.
+
+## Open questions
+
+Not everything has been decided yet: decorators are a very big feature and defining them in the best possible way is hard.
+
+### Where should decorators on exported classes go?
+> [tc39/proposal-decorators#69](https://github.com/tc39/proposal-decorators/issues/69)
+
+The decorator proposal has gone back and forth on this question: should decorators come before or after the export keyword?
+
+```javascript=
+export @decorator class MyClass {}
+
+// or
+
+@decorator
+export class MyClass {}
+```
+
+The underlying question is whether or not the `export` keyword is part of the class declaration or it is a "wrapper". In the first case it should come *after* decorators, since decorators come at the beginning of the declaration; in the second one it should come *before*, because decorators are part of the class declaration.
+
+### How to make decorators securely interact with private elements?
+> [tc39/proposal-decorators#129](https://github.com/tc39/proposal-decorators/issues/129), [tc39/proposal-decorators#133](https://github.com/tc39/proposal-decorators/issues/133)
+
+Decorators give rise to an important security concern: if it is possible to decorate private elements, then private names (which can be considered as the "keys" of private elements) could be leaked. There are different safety levels to be thought of:
+  1) Decorators should not accidentally leak private names. Malicious code should not be able to "steal" private names from other decorators, in any way.
+  2) Only decorators directly applied to private elements might be considered trusted: should class decorators not be able to read and write private elements?
+  3) *Hard privacy* (one of the goals of the class fields proposal) means that private elements should only be accessible from inside the class: should any decorator have access to private names? Should it be only possible to decorate public elements?
+
+These questions need further discussion before being resolved, and that's where Babel comes in.
+
+## The role of Babel
+
+As already stated in the «[What's Happening With the Pipeline (|>) Proposal?](http://babeljs.io/blog/2018/07/19/whats-happening-with-the-pipeline-proposal)» article, with the Babel 7 release we are starting to use our position in the JS ecosystem to help proposal authors even more, by giving developers the ability to test and give feedback about different variations of the proposals.
```

Comment:
Wording tweak: "As already stated" could come off as a bit harsh to people who haven't read the other article; maybe instead, something like "Following the trend in the [_] article,"